src/validate.rs

```diff
@@ -14,3 +14,12 @@ pub fn validate_credential_path(credential_path: &str) {
         ic_cdk::trap("credential path must start with '/' or '?' unless empty");
     }
 }
+
+pub fn validate_credential_headers(credential_headers: &[(String, String)]) {
+    if credential_headers
+        .iter()
+        .any(|(name, _value)| name == "Content-Type")
+    {
+        ic_cdk::trap("unexpected credential header: 'Content-Type'");
```

Comment:
Why is this the only invalid type?
And why does the function trap instead of returning an error?

Reply:
`Content-Type: application/json` is required for JSON-RPC endpoints, so this function makes sure that the user can't accidentally send two different `Content-Type` headers. 

I'll refactor all the validation trap cases in this file to a `ValidationError` enum. The plan was to continue in separate PRs to simplify the code review (currently at ~1000 changed lines of code), although this could be a relatively quick one.